src/uniqueArrayOf.js

```diff
@@ -1,20 +1,58 @@
-import { arrayOf } from 'prop-types';
+import { arrayOf, array } from 'prop-types';
 import and from './and';
 import uniqueArray from './uniqueArray';
 
 const unique = uniqueArray();
 
-export default function uniqueArrayOfTypeValidator(type, name = 'uniqueArrayOfType') {
+export default function uniqueArrayOfTypeValidator(type, ...rest) {
   if (typeof type !== 'function') {
     throw new TypeError('type must be a validator function');
   }
 
+  let mapper = rest.filter(element => typeof element === 'function');
```

Comment:
wouldn't this error out with `rest` being `[mapper, name]` or `[name, mapper]`?

You may want instead to check if the length is 0, 1, or 2, and ensure it's [Function], [String], or [Function, String] only